mlir/lib/Target/LLVMIR/Dialect/OpenMP/OpenMPToLLVMIRTranslation.cpp

```diff
@@ -2496,8 +2496,6 @@ convertDeclareTargetAttr(Operation *op, mlir::omp::DeclareTargetAttr attribute,
       if (declareType == omp::DeclareTargetDeviceType::host) {
         llvm::Function *llvmFunc =
             moduleTranslation.lookupFunction(funcOp.getName());
-        llvmFunc->replaceAllUsesWith(
```

Comment:
I think this code or similar should remain, because otherwise host functions containing target regions calling other host functions containing target regions will give problems. This example should work, and I think this change may have broken it:

```fortran
subroutine before(x)
  integer, intent(out) :: x
  !$omp target map(from: x)
    x = 1
  !$omp end target
end subroutine

subroutine middle()
  integer :: x
  !$omp target map(from: x)
    x = 0
  !$omp end target
  call before(x)
  call after(x)
end subroutine

subroutine after(x)
  integer, intent(out) :: x
  !$omp target map(from:x)
    x = 2
  !$omp end target
end subroutine
```

Reply:
In the functionFiltering, all calls to the host functions are removed regardless of whether they have a target region in them. As such the above test works as is.

Reply:
Ah I see, thanks for clarifying!